From 816fbc8f961b04a72f684c4aa74f240972efa3e9 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 19 Apr 2021 09:40:10 +0800
Subject: [PATCH 49/93] HACK: gl-renderer: Fix random hang during hotplug

Try to bind to a valid output before attaching buffers for mali.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/renderer-gl/gl-renderer.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libweston/renderer-gl/gl-renderer.c b/libweston/renderer-gl/gl-renderer.c
index 51e7078..5be8e85 100644
--- a/libweston/renderer-gl/gl-renderer.c
+++ b/libweston/renderer-gl/gl-renderer.c
@@ -1909,6 +1909,10 @@ gl_renderer_flush_damage(struct weston_surface *surface,
 	texture_used = false;
 	wl_list_for_each(view, &surface->views, surface_link) {
 		if (view->plane == &surface->compositor->primary_plane) {
+			/* HACK: Mali needs a valid context for uploading */
+			if (gr->is_mali_egl && view->output)
+				use_output(view->output);
+
 			texture_used = true;
 			break;
 		}
-- 
2.20.1

